src/keycloak-theme/keycloak-context.provider.ts

```diff
@@ -0,0 +1,146 @@
+      overrides: {},
+      overridesPerPage: {}
+    });
+
+    window.kcContext = getKcContextMock({
+      pageId: environment.mockPageId as LoginThemePageId,
+      overrides: {
+        social: {
+          displayInfo: true,
+          providers: [
+            {
+              loginUrl: "google",
+              alias: "google",
+              providerId: "google",
+              displayName: "Google",
+              iconClasses: "fa fa-google"
+            },
+            {
+              loginUrl: "microsoft",
+              alias: "microsoft",
+              providerId: "microsoft",
+              displayName: "Microsoft",
+              iconClasses: "fa fa-windows"
+            },
+            {
+              loginUrl: "facebook",
+              alias: "facebook",
+              providerId: "facebook",
+              displayName: "Facebook",
+              iconClasses: "fa fa-facebook"
+            },
+            {
+              loginUrl: "instagram",
+              alias: "instagram",
+              providerId: "instagram",
+              displayName: "Instagram",
+              iconClasses: "fa fa-instagram"
+            },
+            {
+              loginUrl: "twitter",
+              alias: "twitter",
+              providerId: "twitter",
+              displayName: "Twitter",
+              iconClasses: "fa fa-twitter"
+            },
+            {
+              loginUrl: "linkedin",
+              alias: "linkedin",
+              providerId: "linkedin",
+              displayName: "LinkedIn",
+              iconClasses: "fa fa-linkedin"
+            },
+            {
+              loginUrl: "stackoverflow",
+              alias: "stackoverflow",
+              providerId: "stackoverflow",
+              displayName: "Stackoverflow",
+              iconClasses: "fa fa-stack-overflow"
+            },
+            {
+              loginUrl: "github",
+              alias: "github",
+              providerId: "github",
+              displayName: "Github",
+              iconClasses: "fa fa-github"
+            },
+            {
+              loginUrl: "gitlab",
+              alias: "gitlab",
+              providerId: "gitlab",
+              displayName: "Gitlab",
+              iconClasses: "fa fa-gitlab"
+            },
+            {
+              loginUrl: "bitbucket",
+              alias: "bitbucket",
+              providerId: "bitbucket",
+              displayName: "Bitbucket",
+              iconClasses: "fa fa-bitbucket"
+            },
+            {
+              loginUrl: "paypal",
+              alias: "paypal",
+              providerId: "paypal",
+              displayName: "PayPal",
+              iconClasses: "fa fa-paypal"
+            },
+            {
+              loginUrl: "openshift",
+              alias: "openshift",
+              providerId: "openshift",
+              displayName: "OpenShift",
+              iconClasses: "fa fa-cloud"
+            }
+          ]
+        }
```

Comment:
This looks a bit hacky to me.  

I can see why you're doing this but I think the base mock should provide the default state of the page.  
It would be nice to have storybook support since what we actually want is to be able to preview the pages in different configurations.

Reply:
I'll try to implement storybook

Reply:
I have already started to do this. Maybe you can take a look when I am finished.

Reply:
ok I started too, I'm struggling with the mock import, but I think it is related to the translation load cause storybook can't find an i18n message

Reply:
// .storybook/main.ts
```
import type { StorybookConfig } from "@storybook/angular";

const config: StorybookConfig = {
  stories: ["../src/**/*.mdx", "../src/**/*.stories.@(js|jsx|mjs|ts|tsx)"],
  addons: [],
  framework: {
    name: "@storybook/angular",
    options: {}
  }
};
export default config;
```

Reply:
.storybook/preview.ts
```
import { applicationConfig, type Preview } from "@storybook/angular";
import { createGetKcContextMock } from "keycloakify/login/KcContext";
import { appConfig } from "../src/keycloak-theme/app.config";
import { kcEnvDefaults, themeNames } from "../src/keycloak-theme/login/models/kc.gen";
import {
  KcContextExtension,
  KcContextExtensionPerPage
} from "../src/keycloak-theme/login/models/KcContext";
const kcContextExtension: KcContextExtension = {
  themeName: themeNames[0],
  properties: {
    ...kcEnvDefaults
  }
};
const kcContextExtensionPerPage: KcContextExtensionPerPage =
  {} as KcContextExtensionPerPage;
const { getKcContextMock } = createGetKcContextMock({
  kcContextExtension,
  kcContextExtensionPerPage,
  overrides: {},
  overridesPerPage: {}
});
const preview: Preview = {
  decorators: [applicationConfig(appConfig)],
  beforeEach: context => {
    context.args["kcContext"] = getKcContextMock({
      pageId: context.globals["pageId"],
      overrides: {
        "x-keycloakify": {},
        locale: { currentLanguageTag: "en" }
      }
    });
  },
  parameters: {
    controls: {
      matchers: {
        color: /(background|color)$/i,
        date: /Date$/i
      }
    }
  }
};

export default preview;
```

Reply:
src/keycloak-theme/login/pages/login/login.stories.ts
```
import { AsyncPipe, NgClass } from "@angular/common";
import { moduleMetadata, type Meta, type StoryObj } from "@storybook/angular/";
import { PasswordWrapperComponent } from "../../common/components/password-wrapper/password-wrapper.component";
import { KcClassPipe } from "../../common/pipes/classname.pipe";
import { SanitizeHtmlPipe } from "../../common/pipes/sanitize-html.pipe";
import { TemplateComponent } from "../../template.component";
import { LoginComponent } from "./login.component";

const meta: Meta<LoginComponent> = {
  globals: {
    pageId: "login.ftl"
  },
  component: LoginComponent,
  decorators: [
    moduleMetadata({
      providers: [],
      imports: [
        KcClassPipe,
        AsyncPipe,
        SanitizeHtmlPipe,
        PasswordWrapperComponent,
        NgClass,
        TemplateComponent
      ]
    })
  ]
};

export default meta;
type Story = StoryObj<LoginComponent>;

export const Primary: Story = {};
```

Reply:
Okay I would prefer we finish the refactoring before integrating storybook.

Reply:
lmk if I can do something to help you!